packages/vite/src/node/config.ts

```diff
@@ -826,56 +831,39 @@ export async function loadConfigFromFile(
 
   let resolvedPath: string | undefined
   let isTS = false
-  let isESM = false
+  let isESM: boolean | 'auto' = 'auto'
   let dependencies: string[] = []
 
-  // check package.json for type: "module" and set `isMjs` to true
-  try {
-    const pkg = lookupFile(configRoot, ['package.json'])
-    if (pkg && JSON.parse(pkg).type === 'module') {
-      isESM = true
-    }
-  } catch (e) {}
-
   if (configFile) {
     // explicit config path is always resolved from cwd
     resolvedPath = path.resolve(configFile)
-    isTS = configFile.endsWith('.ts')
+    isTS = /\.[cm]?ts$/.test(configFile)
 
-    if (configFile.endsWith('.mjs')) {
+    if (/\.m[jt]s$/.test(configFile)) {
       isESM = true
+    } else if (/\.c[jt]s$/.test(configFile)) {
```

Comment:
the judge ignore `js|ts`, Is it intentional? (I think this just need to `else`)

Reply:
It's intentional. The default value of isESM is auto.